### PR TITLE
fix: lemon select clear button was escaping its parent

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

working towards this design https://www.figma.com/file/Y9G24U4r04nEjIDGIEGuKI/PostHog-Design-System-One?node-id=3139%3A6618

the clear button for the lemon select only appears on hover and was very excitable about where it appears

## Changes

* fixes position
* doesn't show only on hover
* uses some CSS important to force override underlying LemonButton

## How did you test this code?

running in storybook and seeing the fix take effect

### CODE REVIEW

File: frontend/src/lib/components/LemonSelect.tsx [hunk 1]
```diff
@@ -134,6 +131,7 @@ export function LemonSelect<O extends LemonSelectOptions>({
                     maxContentWidth: dropdownMaxContentWidth,
                 }}
                 icon={localValue && allOptions[localValue]?.icon}
+                // so that the pop up isn't shown along with the close button
                 sideIcon={isClearButtonShown ? <div /> : undefined}
```
Comment (benjackwhite) [inline frontend/src/lib/components/LemonSelect.tsx:135]: I think there is a generally better fix by somehow merging LemonButtonWithSideAction and LemonButtonWithPopup... I would need to try it myself but I think that would solve all these issues...

**Review by benjackwhite — COMMENTED** [review]
for now I think it's a fine solution and we can try for a cleaner one later 👍

**Review by benjackwhite — APPROVED** [review]

File: frontend/src/lib/components/LemonSelect.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/lib/components/LemonSelect.tsx]: Oh, yep... Am just taking some small steps while I figure out how this all interacts :)

**Comment by pauldambra** [comment]
@clarkus I'll merge because it's a bugfix but I'm still working on this component so feel free to review!

**Pull request merged**